tests/test_lanczos.cc

```diff
@@ -0,0 +1,125 @@
+/*************************************************************************
+ * Copyright (c) 2017-2018 by the mfmg authors                           *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#define BOOST_TEST_MODULE lanczos
+
+#include "mfmg/dealii/lanczos.templates.hpp"
+
+#include <cmath>
+#include <cstdio>
+
+#include "lanczos_simpleop.templates.hpp"
+#include "main.cc"
+
+//-----------------------------------------------------------------------------
+
+BOOST_AUTO_TEST_CASE(lanczos)
+{
+  using namespace mfmg;
+
+  typedef double ScalarType;
+  typedef dealii::Vector<ScalarType> VectorType;
```

Comment:
```suggestion
  typedef dealii::Vector<double> VectorType;
```
Don't bother with `ScalarType`.